ote/src/cljs/ote/views/transport_operator.cljs

```diff
@@ -8,131 +8,431 @@
+       :label (tr [:organization-page :business-id-heading])
+       :type :text-label
+       :full-width? true
+       :h-style :h3}
+
+      {:name ::t-operator/business-id
+       :label ""
+       :element-id "input-business-id"
+       :type :string
+       :validate [[:business-id]]
+       :required? true
+       :warning (tr [:common-texts :required-field])
+       :should-update-check form/always-update
+       :disabled? (ytj-loading? state)
+       :style (if (flags/enabled? :open-ytj-integration)
+                (when (ytj-loading? state) style-base/disabled-control)
+                style-fields/form-field)
+       :on-change #(e! (to/->EnsureUniqueBusinessId %))}
+
+      (when (flags/enabled? :open-ytj-integration)
+        ;; Disabled when business-id is taken or if business-id is not valid or if loading is ongoing
+        {:element-id "btn-submit-business-id"
+         :name ::t-operator/btn-submit-business-id
+         :type :external-button
+         :label (tr [:organization-page :fetch-from-ytj])
+         :primary true
+         :secondary true
+         :on-click #(e! (to/->FetchYtjOperator (::t-operator/business-id operator)))
+         :disabled (or
+                     (empty? (::t-operator/business-id operator))
+                     (not (nil? (get-in state [:transport-operator :ote.ui.form/errors ::t-operator/business-id])))
+                     (:business-id-exists? operator)
+                     (ytj-loading? state))}
+
+        {:name :loading-spinner-ytj
+         :type :loading-spinner
+         :display? (ytj-loading? state)}
+
+        (when ytj-response?
+          (if ytj-success?
+            {:name :ytj-result-msg
+             :type :result-msg-success
+             :content (tr [:organization-page :fetch-from-ytj-success])}
+            {:name :ytj-result-msg
+             :type :result-msg-warning
+             :content (if (= 404 status)
+                        (str (tr [:common-texts :data-not-found])
+                             " " (tr [:common-texts :optionally-fill-manually]))
+                        (str (tr [:common-texts :server-error])
+                             " " (tr [:common-texts :server-error-try-later])
+                             " " (tr [:common-texts :optionally-fill-manually])))})))
+
+      ; label composition for existing business-id
+      (when (get-in state [:transport-operator :business-id-exists?])
+        {:element-id "label-business-id-is-not-unique"
+         :name :business-id-is-not-unique
+         :type :result-msg-warning
+         :content (tr [:common-texts :business-id-is-not-unique])}))))
+
+(defn- operator-form-groups [e! {operator :transport-operator :as state}]
+  "Creates a napote form and resolves data to fields. Assumes expired fields are already filtered from ytj-response."
+  ;(.debug js/console "operator-form-groups: state=" (clj->js state))
+  (let [response-ok? (= 200 (get-in state [:ytj-response :status]))
```

Comment:
response-ok? would be easier to understand if with different name. Like ytj-response-ok? or something.

Reply:
Yes